Source/core/StateTrigger.h

```diff
@@ -110,7 +110,7 @@ namespace Core {
         m_AdminLock.Lock();
 
         // Now wait till our state is reached.
-        while (((a_State & m_State) == 0) && (blTimeOut == false)) {
+        while (((a_State & static_cast<uint32_t>(m_State)) == 0) && (blTimeOut == false)) {
```

Comment:
Although I guess uint32_t and unsigned int are the same at the moment to prevent any problems and cast to the exact same type "unsigned int" (also below of course).